src/lib.ts

```diff
@@ -44,7 +44,8 @@ export default async function isMergable(opts: Options): Promise<Result> {
 			repo: opts.repo,
 			pull_number: opts.pullRequest,
 		}),
-		octokit.repos.listStatusesForRef({
+		// this is not ideal, but their  type definition does not allow for passing the object notation
+		octokit.paginate("GET /repos/:owner/:repo/statuses/:ref", {
```

Comment:
Does this accept a type parameter instead of casting `statuses as` below?

Reply:
No it does not. I tried to update the octokit library, but that broke typedefs even more. Since this is a bit more stable I can take a second look at that upgrade.

Reply:
err v18 is like super different, this could take some digging.

Reply:
not terribly important, happy to merge this

Reply:
I actually got the upgrade completed, but it has some hacky types #5